qadence/constructors/feature_maps.py

```diff
@@ -1,106 +1,224 @@
 from __future__ import annotations
 
-from typing import Type, Union
+import inspect
+import math
+from typing import Callable, Type, Union
 
-import numpy as np
 import sympy
```

Comment:
Since you seem to only use `sympy.Functions` and `sympy.pi` in this module, I'd import them to avoid importing the whole library.